File: src/python/PythonSDK/foundationallm/langchain/data_sources/cxo/cxo_configuration.py
from typing import List, Literal
from foundationallm.langchain.data_sources import DataSourceConfiguration
from foundationallm.langchain.data_sources.search_service import SearchServiceConfiguration

class CXOConfiguration(SearchServiceConfiguration):

    """Configuration settings for a stock agent"""
    configuration_type: Literal['cxo']
    sources : List[str]
    retriever_mode : str
    company : str

File: src/dotnet/AgentFactory/Models/Orchestration/DataSourceConfigurations/CXOConfiguration.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FoundationaLLM.AgentFactory.Core.Models.Orchestration.DataSourceConfigurations
{
    /// <summary>
    /// Configuration for the CXO agent.
    /// </summary>
    public class CXOConfiguration : SearchServiceConfiguration
    {
        /// <summary>
        /// The type of configuration. This value should not be changed.
        /// </summary>
        [JsonProperty("configuration_type")]
        public new string ConfigurationType = "cxo";
        /// <summary>
        /// Search filter elements.
        /// </summary>
        [JsonProperty("sources")]
        public string[]? Sources { get; set; }

        /// <summary>
        /// The vector database.
        /// </summary>
        [JsonProperty("retriever_mode")]
        public required string RetrieverMode { get; set; }

        /// <summary>
        /// The name of the CXO's company.
        /// </summary>
        [JsonProperty("company")]
        public required string Company { get; set; }
    }
}
